src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -37,37 +39,77 @@
 
   ParameterizedTypeName(ClassName rawType, List<TypeName> typeArguments,
       List<AnnotationSpec> annotations) {
+    this(rawType, null, typeArguments, annotations);
+  }
+
+  private ParameterizedTypeName(ClassName rawType, ParameterizedTypeName enclosingType,
```

Comment:
nit: it’s weird that the arguments are (rawType, enclosingType) here, and the opposite in field declarations